lib/DDG/Rewrite.pm

```diff
@@ -156,8 +156,11 @@ sub _build_nginx_conf {
 
 	$cfg .= "\techo_before_body '".$self->callback."(';\n" if $wrap_jsonp_callback;
 	$cfg .= "\techo_before_body '".$self->callback.qq|("';\n| if $wrap_string_callback;
+	my ($spice_name) = $self->path =~ /js\/spice\/(.+?)\//;
```

Comment:
They do.  You might want to lift the IA name derivation from DDG::FileCache::_sharedir_to_name and tweak it.